### DISCUSSION TITLE
Why is the current clue text constrained to 481px?

### DISCUSSION THREAD

**Discussion by DKMiller71** [viresh-ratnakar/exolve#101] [Q&A]
I've been wondering: why does the current clue text seem to be constrained to 481px?  On my page and other users' pages, it seems that the clue area is scrunched up (and wrapping) to fit over the grid instead of being allowed to use the horizontal area across the page (such as over the columns of clues) if the page is wider. Is there a reason for this?

I see that if I tinker with its width, it takes away the space of the columns of clues, so it might be the nesting order, but it had me wondering.

**Comment by viresh-ratnakar** [comment]
On phones, and in most cases in blogs (that carry the crossword in the main central column and other stuff on either side), only the grid-width is reliably visible. So I've opted to keep the code simple: the current-clue div is nested within the grid-panel. Iirc, at one point I had played around with changing this (indeed to use the full width when available), but was somehow not satisfied with what I got.

All that said, I will maybe give it another shot, as I do not remember exactly what I ran into.

Can you point me to some example puzzles/clues where you see the clue area as "scrunched up"? I looked at [this one](https://dkmiller71.github.io/DarrensPuzzles/puzzles/xword/Cryptic202404) but it looked OK to me (Chrome/Linux).

**Reply by DKMiller71** [reply]
Like this, for example...

![image](https://github.com/viresh-ratnakar/exolve/assets/45632749/545fccf3-41f5-4101-bd07-750095482284)

**Reply by DKMiller71** [reply]
Experimenting, I made the following changes in my local copy (attached, search on DKM): 

Moved lines 584-589 (`${this.prefix}-curr-clue` and its children) up to below 579 (`${this.prefix}-clear-area`)

Changed line 5120:
`this.currClue.style.width = '100%'; // widthPx;`

This seemed to work, even if I changed the browser to mobile view or load the page into an iframe. But I don't know what other scenarios you might have in mind.
[exolve-m.js.txt](https://github.com/viresh-ratnakar/exolve/files/15311321/exolve-m.js.txt)

**Comment by viresh-ratnakar** [comment]
Thanks, I'll take a look soon (it might take me a couple of days to get to
this).

- If I do do this (and chances are, I will, why not), I will add this as an
*option* to opt into. I myself find the current layout quite acceptable
(the word "scrunched" did not spring to my mind when I looked at the image
that you attached :-)), but it's indeed possible that I'm just too used to
it.
- The other main thing I need to check is that this does not break Exet.
Even if it does, I'm sure that's fixable, and I'll make any necessary
tweaks to Exet as well.

**Reply by DKMiller71** [reply]
That's totally fine - whatever you think best. I have a workaround anyway....

inside `customizeExolve(puz)`:
```
	puz.clearArea.after(puz.currClue);
	puz.resizeCurrClueAndControls = customResizeCurrClueAndControls;
```

Where `customResizeCurrClueAndControls` is just a copy of the original code with the replacement line:
` this.currClue.style.width = '100%';`
